### PR TITLE
Moth wings can now open

### PR DESCRIPTION
**Pull request by red031000**
## About The Pull Request

ports BeeStation/NSV13#1858
which in turn is a partial port of yogstation13/Yogstation#11040

Adds the ability for moths to open/extend their wings
In addition, adjusts the timings for flap and aflap to be half the time, as they did not seem fast enough

To prevent spam, and having multi-tile wings on roundstart, the wing emote is now only available after an upgrade to the moth's wing's flight level to flying or higher. flap and aflap is unaffected

## Why It's Good For The Game

Adds flavour to moths, while also looking nice

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

unknown

unknown

108682581-c621a980-74f0-11eb-8a61-2a84cfadc8e2

</details>

## Changelog
:cl:
add: Add the ability for moths to open their wings when they are able to fly (either through genetics, or flight potions)
tweak: Change time for flap and aflap to half
imageadd: Add moth wings open images from yogstation
/:cl:

### CODE REVIEW

**Comment by Karmic-Skink** [comment]
Honestly these opening up to span across multiple tiles is a bit much, most carbon sprites are contained within 32x32.

**Comment by Kapu1178** [comment]
more reason to add organ rendering instead of relying on this ancient dog ass sprite accessory snowflake mess

**Comment by Fronsis** [comment]
Please next thing for moths to port needs to be the *chitters sound (and maybe port over some others for the other species)

Really cool! but i'll let people like Kapu judge this stuff, just wanted to push my *chitters agenda

**Comment by Crossedfall** [comment]
> Honestly these opening up to span across multiple tiles is a bit much, most carbon sprites are contained within 32x32.

Inclined to agree. The sprites need to be altered so they fit within a smaller space.

**Review by Crossedfall — DISMISSED** [review]
> Honestly these opening up to span across multiple tiles is a bit much, most carbon sprites are contained within 32x32.

Inclined to agree. The sprites need to be altered so they fit within a smaller space.

**Comment by red031000** [comment]
I'm not a spriter myself, I didn't make these, I just ported it directly from yogstation, as such I wouldn't have any idea how to fix this

**Comment by red031000** [comment]
plus, these are already a similar size to the open wings that already exist within the codebase (see: dragon, angel)

**Comment by Crossedfall** [comment]
> I'm not a spriter myself, I didn't make these, I just ported it directly from yogstation, as such I wouldn't have any idea how to fix this

Bring it up in our spriting channel, see if anyone would be willing to either walk you through the process or make the edits for you.

> plus, these are already a similar size to the open wings that already exist within the codebase (see: dragon, angel)

They're not stock carbon "parts" and are substantially rarer. The wingspans of those are considered to be acceptable exemptions.

**Comment by Bokkiewokkie** [comment]
> > plus, these are already a similar size to the open wings that already exist within the codebase (see: dragon, angel)
> 
> They're not stock carbon "parts" and are substantially rarer. The wingspans of those are considered to be acceptable exemptions.

Definitely agree, though you could consider letting moths have these wings when they can actually open their wings to fly? (after drinking a flight potion moth wings can fly too.)

**Comment by red031000** [comment]
since a lot of different opinions have been expressed, what direction should I go with this?

**Comment by red031000** [comment]
what I could do is change the *wing command to open open the wings if they can be used for atmosphere flight, that should alleviate the main issue with large sprites on round spawn, but still have the sprites if (through either mining or rainbow slime) they get a flight potion

**Comment by zeskorion** [comment]
Why does this use moth_wingsopen? Why not make it use normal wing bodyparts?

**Comment by red031000** [comment]
the moth wings are in a separate file from the normal wings, plus they are reliant on the moth wings bodypart

**Review by PowerfulBacon — COMMENTED** [review]
Not a mega fan of moth wings and normal wings being seperated, since it implies you can have wings & moth wings at the same time.
I assume the reason for this is because of the code in emote.dm
```
if(H.dna && H.dna.species)
		if(H.dna.features["wings"] != "None")
			return TRUE
		if(H.dna.features["moth_wings"] != "None")
			var/obj/item/organ/wings/wings = H.getorganslot(ORGAN_SLOT_WINGS)
			if(H.getorgan(/obj/item/organ/wings))
				if(wings.flight_level >= WINGS_FLYING)
					return TRUE
```
I would suggest adding a variable to regular wings which is a minimum flight level to open the wings to full size and then overriding it on moth wings to be WINGS_FLYING

**Comment by red031000** [comment]
@PowerfulBacon how would you like it to be merged? currently they both take up the same organ slot so you can't actually have both at the same time, plus I can see this creating a potential problem for character creation, as if they are classified the same, and pull from the same list, there'd need to be a manual list on what is acceptable for creation, and it'd still be (somewhat) separate

**Comment by PowerfulBacon** [comment]
Right, I forgot how bad character customisation code is

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -145,8 +145,14 @@
...
 	var/mob/living/carbon/human/H = user
-	if(H.dna && H.dna.species && (H.dna.features["wings"] != "None"))
-		return TRUE
+	if(H.dna && H.dna.species)
+		if(H.dna.features["wings"] != "None")
+			return TRUE
+		if(H.dna.features["moth_wings"] != "None")
+			var/obj/item/organ/wings/wings = H.getorganslot(ORGAN_SLOT_WINGS)
+			if(H.getorgan(/obj/item/organ/wings))
+				if(wings.flight_level >= WINGS_FLYING)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/emote.dm]: Could this not just be
if(istype(wings)) rather than getting the organ twice?

**Review by PowerfulBacon — APPROVED** [review]
I hate species code, but theres not much that this PR can do about that

**Pull request merged**